File: ejpcc-client/src/api/AccountingService.js
import axios from "axios";
class AccountingService {
  constructor() {
    this.service = axios.create({
      baseURL: `${process.env.VUE_APP_BACKEND_URL}/`,
    });
  }

  fetchAll(drawId) {
    return this.service.get(`draws/${drawId}/accountings`);
  }

  startAccounting(draw) {
    return this.service.post(`/draws/${draw.id}/accountings/actions`, {
      type: "CALCULATE_ACCOUNTINGS",
    });
  }

  approve(accountingValues) {
    return this.service.post(`/draws/${accountingValues.draw.id}/accountings/actions`, {
      type: "CONFIRM_ACCOUNTING_CALCULATION",
    });
  }
}

export default new AccountingService();

File: accountings-service/src/main/java/de/westlotto/accountings/service/AccountingService.java
package de.westlotto.accountings.service;

import de.westlotto.accountings.model.AccountingValues;
import de.westlotto.accountings.model.CompanyAccountingValues;

public interface AccountingService {

    AccountingValues findAccountingValues(Long drawId);

    CompanyAccountingValues findCompanyAccountingValues(Long drawId, Long companyId);

    AccountingValues calculateAccountingValues(Long drawId);

    AccountingValues approve(Long drawId);
}
